tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/Init.java

```diff
@@ -89,16 +89,18 @@ public static Init create(Scope scope) {
    *
    * <p>Registered initializers are then grouped as a single unit of computation by adding
    * and executing an {@link org.tensorflow.op.core.Init#create(Scope) init} operation from a graph
-   * session.
+   * session.  No-ops if the session is eager.
    *
    * @param scope
    * @param initializer
-   * @throws IllegalArgumentException if the execution environment in scope is not a graph
    * @see org.tensorflow.op.core.Init#create(Scope) init
    */
   @Endpoint(name = "initAdd")
   public static void add(Scope scope, Op initializer) {
     ExecutionEnvironment exEnv = scope.env();
+    if(exEnv.isEager()){
+      return;
```

Comment:
Why leave the exception in there if it's not triggered?

Reply:
Because in theory you could have a non-eager execution environment that isn't Graph.  Is there any possibility of that being a thing?

Reply:
Not to my knowledge. There are a lot of other things that assume either eager or graph execution and that would blow up if there was a third type. At the very least the exception error message should be fixed to note that this op succeeds on graph and eager sessions.